lib/gears/string.lua

```diff
@@ -0,0 +1,86 @@
+---------------------------------------------------------------------------
+--- String module for gears
+--
+-- @module gears.string
+---------------------------------------------------------------------------
+
+local gstring = {}
+
+local xml_entity_names = { ["'"] = "&apos;", ["\""] = "&quot;", ["<"] = "&lt;", [">"] = "&gt;", ["&"] = "&amp;" };
+--- Escape a string from XML char.
+-- Useful to set raw text in textbox.
+-- @class function
+-- @name escape
+-- @param text Text to escape.
+-- @return Escape text.
+function gstring.escape(text)
+    return text and text:gsub("['&<>\"]", xml_entity_names) or nil
+end
+
+local xml_entity_chars = { lt = "<", gt = ">", nbsp = " ", quot = "\"", apos = "'", ndash = "-", mdash = "-", amp = "&" };
+--- Unescape a string from entities.
```

Comment:
The more I think of pros/cons the more I agree with this.

@Elv13 what's your say on this? `escape` is used in the tasklist, taglist, prompt, menu, and menubar (from what I can tell). The name change wouldn't be hit until we actually remove it from `awful.util`. By then we'd have updated these to the new `gears.string` functions.